### DISCUSSION TITLE
Best async worker

### DISCUSSION THREAD

**Discussion by RobinFrcd** [fastapi/fastapi#5106] [Questions] [answered]
I'm currently running `async` endpoints which are themselves asynchronously calling long-running APIs.

I'm running FastAPI with `uvicorn.workers.UvicornWorker` workers, but I've seen in multiple (old) [blog posts](https://medium.com/@danieldng/use-gevent-when-your-gunicorn-worker-is-waiting-for-data-180efef96367) advising to use async workers like [gevent](http://www.gevent.org/).

But when look at [FastAPi doc](https://fastapi.tiangolo.com/async/#other-forms-of-asynchronous-code), looks like these workers are a bit deprecated.

So my question is, what are the best workers for FastAPI when running long-running async requests ?

Thanks

**Comment by panla** [comment]
## server and worker

### 1 gunicorn

```bash
gunicorn server:app -k uvicorn.workers.UvicornWorker ......
```

ps: it is not all asgi

### 2 uvicorn

```bash
uvicorn server:app --loop uvloop --http httptools ....
```

### 3 Daphne

```bash
daphne app:App ...
```

### 4 Hypercorn

```bash
hypercorn app:App
```

**Reply by RobinFrcd** [reply]
Thank you, but is one of these more suited to handle async requests ?

**Reply by iudeen** [reply, marked as answer]
Uvicorn is one of the fastest ASGI servers. If you want H2 support, you can use Hypercorn

**Reply by sergenti** [reply]
Hey there, @iudeen; I'm having the same question as @RobinFrcd.

any chance you can explain why this is important` --loop uvloop --http httptools`? I haven't seen it in other implementations

**Reply by sergenti** [reply]
If these are all the configs, what makes the fastest uvicorn worker? 
https://www.uvicorn.org/#usage

**Reply by sergenti** [reply]
Also, I was reading this one 
![image](https://github.com/tiangolo/fastapi/assets/55017307/2fb41488-3909-44dc-a8da-769d4444158a)

https://www.uvicorn.org/#running-with-gunicorn
https://www.uvicorn.org/deployment/
